.github/workflows/build-pr.yml

```diff
@@ -69,6 +69,7 @@ jobs:
     permissions:
       id-token: write
       contents: read
+    environment: aks-deploy
```

Comment:
This was mentioned during the Meeting, could you remind me how the environment property is used?

Reply:
without the env, will the cached token be accessible?

Reply:
Sure. Environment are a github construct to represent a target deployment type. So testing, dev, prod, etc. In the environment, you can setup workflows, secrets, and variables that target that environment. So in our case, we would define a new environement `azure-test` and this environment contains all the necessary creds for our azure e2e tests. Read more about environments [here](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment).

In our workflows, we use `Azure/login` [action](https://learn.microsoft.com/en-us/entra/workload-id/workload-identity-federation-create-trust?pivots=identity-wif-apps-methods-azp#environment-example) to auth. The federated credential accepts `environment` as one of the scopes for federated credential. This means for each unique `<org>/ratify` combination, we could setup one federated credential that will target an environment `azure-test`. This means that for `deislabs` we'd have one cred and for each person's fork we would have another. There are other scopes supported for federated credentials like branches however it becomes cumbersome if we want to manually trigger tests from a specific feature branch. This is why environment is helpful.

Now there is some downside (potentially) for using environment. Now, the new environment will show up on the main github repository name. 
![image](https://github.com/deislabs/ratify/assets/12980641/9a1d5a55-31eb-4212-aea3-5c1cc13d2763)

And what's unfortunate is that for each aks run, it generates 6 entries under `Deployments` and if there are failures in the `latest` one (this seems arbitrary what is latest of the matrix), then the deployment shows as failed. 

![image](https://github.com/deislabs/ratify/assets/12980641/05b0841d-137b-44ca-a04b-9e17196a6f20)

Reply:
So in conclusion we have 2 options for federated credential scope. 

1. Use branch scope. This means it'll be very cumbersome to trigger AKS tests directly from a feature branch. We would need to settle on the `dev` branch to run all tests on your fork.
2. Use environments with the failure reporting following the screenshots above.